pkg/pipeline/encode/opentelemetry/encode_otlpmetrics.go

```diff
@@ -77,30 +75,36 @@ func createKey(name string, keys []string) string {
 }
 
 func (e *EncodeOtlpMetrics) ProcessGauge(m interface{}, name string, labels map[string]string, value float64, lvs []string) error {
-	obs := m.(Float64Gauge)
-	// set attributes using the labels
-	attributes := obtainAttributesFromLabels(labels)
-	key := createKey(name, lvs)
-	obs.Set(key, value, attributes)
-	return nil
+	if obs, ok := m.(Float64Gauge); ok {
+		// set attributes using the labels
+		attributes := obtainAttributesFromLabels(labels)
+		key := createKey(name, lvs)
+		obs.Set(key, value, attributes)
+		return nil
+	}
+	return fmt.Errorf("wrong Otlp Gauge type for %s: %T; expecting Float64Gauge", name, m)
 }
 
-func (e *EncodeOtlpMetrics) ProcessHist(m interface{}, labels map[string]string, value float64) error {
-	histo := m.(metric.Float64Histogram)
-	// set attributes using the labels
-	attributes := obtainAttributesFromLabels(labels)
-	histo.Record(e.ctx, value, metric.WithAttributes(attributes...))
-	return nil
+func (e *EncodeOtlpMetrics) ProcessHist(m interface{}, name string, labels map[string]string, value float64) error {
+	if histo, ok := m.(metric.Float64Histogram); ok {
+		// set attributes using the labels
+		attributes := obtainAttributesFromLabels(labels)
+		histo.Record(e.ctx, value, metric.WithAttributes(attributes...))
+		return nil
+	}
+	return fmt.Errorf("wrong Otlp Histogram type for %s: %T; expecting Float64Histogram", name, m)
 }
 
-func (e *EncodeOtlpMetrics) ProcessAggHist(m interface{}, labels map[string]string, values []float64) error {
-	histo := m.(metric.Float64Histogram)
-	// set attributes using the labels
-	attributes := obtainAttributesFromLabels(labels)
-	for _, v := range values {
-		histo.Record(e.ctx, v, metric.WithAttributes(attributes...))
+func (e *EncodeOtlpMetrics) ProcessAggHist(m interface{}, name string, labels map[string]string, values []float64) error {
+	if histo, ok := m.(metric.Float64Histogram); ok {
+		// set attributes using the labels
+		attributes := obtainAttributesFromLabels(labels)
+		for _, v := range values {
+			histo.Record(e.ctx, v, metric.WithAttributes(attributes...))
+		}
+		return nil
 	}
-	return nil
+	return fmt.Errorf("wrong Otlp Histogram type for %s: %T; expecting Float64Histogram", name, m)
```

Comment:
(minor) Change the message so it's not exactly the same as line 95, such as:
`return fmt.Errorf("wrong Otlp Aggregated Histogram type for %s: %T; expecting Float64Histogram", name, m)`